akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -778,19 +778,22 @@ private[akka] class ShardRegion(
   }
 
   def receiveCommand(cmd: ShardRegionCommand): Unit = cmd match {
+
     case Retry =>
-      sendGracefulShutdownToCoordinator()
+      if (gracefulShutdownInProgress) {
+        sendGracefulShutdownToCoordinator()
+        tryCompleteGracefulShutdown()
+      } else {
```

Comment:
I think this `else` is incorrect. Now it's not registering to the coordinator, which might be needed even if gracefulShutdownInProgress. Otherwise the coordinator will ignore the GracefulShutdownReq